### PR TITLE
[stable8] Fix sharing add to group 8.0

### PR DESCRIPTION
**Pull request by schiessle**
combined backport of https://github.com/owncloud/core/pull/17381 and https://github.com/owncloud/core/pull/18876

I had to apply parts of the backport manually because the code differs. @nickvergessen can you test it, I think you know best how to test it because of all your testing the last few days.

cc @MorrisJobke @karlitschek @cmonteroluque

### CODE REVIEW

File: apps/files_sharing/lib/sharedmount.php [hunk 1]
```diff
@@ -20,17 +20,21 @@ class SharedMount extends MountPoint implements MoveableMount {
...
 
+	/** @var  string */
+	protected $user;
+
 	public function __construct($storage, $mountpoint, $arguments = null, $loader = null) {
+		$this->user = $arguments['user'];
 		// first update the mount point before creating the parent
-		$newMountPoint = $this->verifyMountPoint($arguments['share'], $arguments['user']);
-		$absMountPoint = '/' . $arguments['user'] . '/files' . $newMountPoint;
+		$newMountPoint = $this->verifyMountPoint($arguments['share'], $this->user);
```
Comment (MorrisJobke) [inline apps/files_sharing/lib/sharedmount.php]: `$this->user` is not needed here

**Comment by MorrisJobke** [comment]
https://github.com/owncloud/core/pull/18903#issuecomment-138681000 ?

**Comment by schiesbn** [comment]
@MorrisJobke from looking at the code it looks like we do the same mistake in 8.0. But we should test it and only apply it if it makes a difference.

**Comment by MorrisJobke** [comment]
Generally speaking: We don't have seen any issues with this on 8.0 instances. I would not backport this rather risky change. Having this fixed in 8.1 is enough from my POV

**Comment by schiesbn** [comment]
Let's see if I can reproduce https://github.com/owncloud/core/pull/17381 on 8.0. If I can we backport, if not we close it

**Comment by schiesbn** [comment]
Indeed, it seems like I can't reproduce any of the misbehaviors. So while the fix is still correct it seems to haver no impact on 8.0 -> close

**Pull request closed without merging**